### ISSUE TITLE
[BUG] function `j_loader` does not properly raise when input file non-existing and not with expected extension name

### ISSUE THREAD

**Issue by kirk0830** [deepmodeling/deepmd-kit#4490]
### Bug summary

```python
def j_loader(filename: Union[str, Path]) -> dict[str, Any]:
    """Load yaml or json settings file.

    Parameters
    ----------
    filename : Union[str, Path]
        path to file

    Returns
    -------
    dict[str, Any]
        loaded dictionary

    Raises
    ------
    TypeError
        if the supplied file is of unsupported type
    """
    filepath = Path(filename)
    if filepath.suffix.endswith("json"):
        with filepath.open() as fp:
            return json.load(fp)
    elif filepath.suffix.endswith(("yml", "yaml")):
        with filepath.open() as fp:
            return yaml.safe_load(fp)
    else:
        raise TypeError("config file must be json, or yaml/yml")
```
If the file is not found, it will raise TypeError, but this is not the expected behavior: in this case, should raise FileNotFoundError instead.

### DeePMD-kit Version

devel

### Backend and its version

any

### How did you download the software?

Built from source

### Input Files, Running Commands, Error Log, etc.

no

### Steps to Reproduce

can be easily trigger the inappropriate raise if run deepmd-kit like
```bash
dp train SomeNonExistingFile.OtherThanJsonOrYaml
```

**Comment by njzjz** [comment]
I don't get the point. I feel that both `TypeError` and `FileNotFoundError` are accepted, as there are two errors at the same time, and Python only allows to throw one.
